easybuild/toolchains/compiler/gcc.py

```diff
@@ -110,3 +117,43 @@ def _set_compiler_vars(self):
                 raise EasyBuildError("Failed to determine software root for GCC")
 
         self.variables.append_subdirs("LDFLAGS", gcc_root, subdirs=["lib64", "lib"])
+
+    def _set_optimal_architecture(self, default_optarch=None):
+        """
+        GCC-specific adjustments for optimal architecture flags.
+
+        :param default_optarch: default value to use for optarch, rather than using default value based on architecture
+                                (--optarch and --optarch=GENERIC still override this value)
+        """
+        if default_optarch is None and self.arch == systemtools.AARCH64:
+            # On AArch64, -mcpu=native is not supported prior to GCC 6.  In this case, try to optimize for the detected
+            # CPU model (vanilla ARM cores only).  Note that this heuristic may fail if the CPU model is not supported
+            # by the GCC version being used.
+            gcc_version = self.get_software_version(self.COMPILER_MODULE_NAME)[0]
```

Comment:
@boegel Just to clarify: You are proposing to use `get_software_version` from `easybuild.tools.modules` here rather than `self.get_software_version`, right? I don't think I understand the difference right now (it's been a long day...)